api/circleci.ts

```diff
@@ -1,40 +1,38 @@
-import qs from 'querystring'
 import got from '../libs/got'
 import { createBadgenHandler, PathArgs } from '../libs/create-badgen-handler'
 
+const CIRCLECI_API_URL = 'https://circleci.com/api/v1.1/'
+
+const client = got.extend({ prefixUrl: CIRCLECI_API_URL })
+
 export default createBadgenHandler({
   title: 'CircleCI',
   examples: {
     '/circleci/github/nuxt/nuxt.js': 'build',
     '/circleci/github/nuxt/nuxt.js/master': 'build (branch)',
   },
   handlers: {
-    '/circleci/:vcs<github|gitlab>/:user/:project/:branch?': handler
+    '/circleci/:vcs<github|gitlab>/:owner/:repo/:branch?': handler
   }
 })
 
-async function handler ({ vcs, user, project, branch }: PathArgs) {
-  // https://circleci.com/docs/api/v1-reference/
-  branch = branch ? `/tree/${qs.escape(branch)}` : ''
-  const endpoint = `https://circleci.com/api/v1.1/project/${vcs}/${user}/${project}${branch}?filter=completed&limit=1`
-  const [latest] = await got(endpoint).json<any>()
+async function handler ({ vcs, owner, repo, branch }: PathArgs) {
+  // https://circleci.com/docs/api/#recent-builds-for-a-single-project
+  const path = [
+    `project/${vcs}/${owner}/${repo}`,
+    branch && `tree/${encodeURIComponent(branch)}`
+  ].filter(Boolean).join('/')
```

Comment:
Prefer the previous approach, more straightforward compare to `.filter(Boolean).join('/')`:

```js
const branchPath = branch ? `/tree/${encodeURIComponent(branch)}` : ''
const path = `project/${vcs}/${owner}/${repo}${branch}`
```

Reply:
Reverted :heavy_check_mark: